tracer/encoder.go

```diff
@@ -3,63 +3,138 @@ package tracer
 import (
 	"bytes"
 	"encoding/json"
+
+	"github.com/ugorji/go/codec"
 )
 
-// jsonEncoder encodes a list of spans in JSON format.
+// Encoder is a generic interface that expects an Encode() method
+// for the encoding process, and a Read() method that will be used
+// by the http handler
+type Encoder interface {
+	Encode(traces [][]*Span) error
+	Read(p []byte) (int, error)
+	ContentType() string
+}
+
+var mh codec.MsgpackHandle
+
+// msgpackEncoder encodes a list of traces in Msgpack format
+type msgpackEncoder struct {
+	buffer      *bytes.Buffer
+	encoder     *codec.Encoder
+	contentType string
+}
+
+func newMsgpackEncoder() *msgpackEncoder {
+	buffer := &bytes.Buffer{}
+	encoder := codec.NewEncoder(buffer, &mh)
+
+	return &msgpackEncoder{
+		buffer:      buffer,
+		encoder:     encoder,
+		contentType: "application/msgpack",
+	}
+}
+
+// Encode serializes the given traces list into the internal
+// buffer, returning the error if any
+func (e *msgpackEncoder) Encode(traces [][]*Span) error {
+	e.buffer.Reset()
```

Comment:
Oh actually we're only using the ``Encode()`` and not the ``Decode()`` because it's supposed that the client only encodes data. I used ``Decode()`` in tests just to be sure that the payload was properly encoded (unittest that is also covered in the integration tests).

I think we can document these things later.